athena-elasticsearch/src/main/java/com/amazonaws/connectors/athena/elasticsearch/ElasticsearchSchemaUtils.java

```diff
@@ -0,0 +1,252 @@
+     */
+    private FieldType toFieldType(LinkedHashMap<String, Object> mapping)
+    {
+        logger.info("toFieldType - enter: " + mapping);
+
+        String elasticType = (String) mapping.get("type");
+        Types.MinorType minorType;
+        Map<String, String> metadata = new HashMap<>();
+
+        switch (elasticType) {
+            case "text":
+            case "keyword":
+            case "binary":
+                minorType = Types.MinorType.VARCHAR;
+                break;
+            case "long":
+                minorType = Types.MinorType.BIGINT;
+                break;
+            case "integer":
+                minorType = Types.MinorType.INT;
+                break;
+            case "short":
+                minorType = Types.MinorType.SMALLINT;
+                break;
+            case "byte":
+                minorType = Types.MinorType.TINYINT;
+                break;
+            case "double":
+                minorType = Types.MinorType.FLOAT8;
+                break;
+            case "scaled_float":
+                minorType = Types.MinorType.BIGINT;
+                metadata.put("scaling_factor", mapping.get("scaling_factor").toString());
+                break;
+            case "float":
+            case "half_float":
+                minorType = Types.MinorType.FLOAT4;
+                break;
+            case "date":
+            case "date_nanos":
+                minorType = Types.MinorType.DATEMILLI;
+                break;
+            case "boolean":
+                minorType = Types.MinorType.BIT;
+                break;
+            default:
+                minorType = Types.MinorType.NULL;
+                break;
+        }
+
+        logger.info("Arrow Type: {}, metadata: {}", minorType.toString(), metadata);
+
+        return new FieldType(true, minorType.getType(), null, metadata);
+    }
+
+    /**
+     * Checks that two mappings are equal irrespective of the ordering of the fields and their children inside
+     * of the Schema objects.
+     * @param mapping1 is a mapping to be compared.
+     * @param mapping2 is a mapping to be compared.
+     * @return true if the lists are equal, false otherwise.
+     */
+    protected final boolean mappingsEqual(Schema mapping1, Schema mapping2)
+    {
+        logger.info("mappingsEqual - Enter:\nMapping1: {}\nMapping2: {}", mapping1, mapping2);
+
+        // Schemas must have the same number of elements.
+        if (mapping1.getFields().size() != mapping2.getFields().size()) {
+            logger.warn("Mappings are different sizes!");
+            return false;
+        }
+
+        // Mappings must have the same fields (irrespective of internal ordering).
+        for (Field field1 : mapping1.getFields()) {
+            Field field2 = mapping2.findField(field1.getName());
+            if (field2 == null || field1.getType() != field2.getType()) {
+                logger.warn("Mapping fields mismatch!");
```

Comment:
what does this actually mean, why is it bad? also would be useful to log more info about what happened (which fileds, what was the mistmatch).

Reply:
Added comment and enhanced log.